platform/linux/mod.rs

```diff
@@ -30,7 +30,8 @@ const MAP_FAILED: *mut u8 = (!0usize) as *mut u8;
 
 // The value Linux returns for SO_SNDBUF
 // is not the size we are actually allowed to use...
-const RESERVED_SIZE: usize = 256;
+// Empirically, we have to deduct 32 bytes from that.
+const RESERVED_SIZE: usize = 32;
```

Comment:
Indeed.

The man page actually mentions the buffer being *twice* the usable size, but in reference to a different socket type. There is no documentation about this for `SOCK_SEQPACKET` as far as I'm aware.